### PR TITLE
Wait for secret deletion in the stabilize operation of namespace delete handler

### PR DESCRIPTION
**Pull request by nidhimanthale**
*Issue #, if available:*

*Description of changes:*
- Remove 30 seconds wait after namespace deletion in namespace delete handler 
- Add secrets manager dependency and client
- Modify the stabilize operation for namespace delete handler to verify that the secret associated with the namespace is deleted 

[Testing]
- Unit Tests
- Manually tested that deletion of following namespaces followed expected behaviour
    - Namespace without secret -> Deleted successfully
    - Namespace with secret and default key -> Deleted successfully
    - Namespace with secret and customer managed key -> Deleted successfully
    - Above three cases with workgroup attached. Delete workgroup and then namespace -> Deleted successfully 
    - Invalid Namespace -> Returns NotFound Error

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-redshiftserverless-namespace/src/main/java/software/amazon/redshiftserverless/namespace/ReadHandler.java [hunk 1]
```diff
@@ -31,6 +32,7 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
         final CallbackContext callbackContext,
         final ProxyClient<RedshiftServerlessClient> proxyClient,
         final ProxyClient<RedshiftClient> redshiftProxyClient,
+        final ProxyClient<SecretsManagerClient> secretsManagerProxyClient,
```
Comment (vramanet) [inline aws-redshiftserverless-namespace/src/main/java/software/amazon/redshiftserverless/namespace/ReadHandler.java:35]: Why is this needed here ?

**Review by vramanet — COMMENTED** [review]
Looks good to me. Approved

**Review by mashuaihh — APPROVED** [review]

**Review by vramanet — APPROVED** [review]
